### ISSUE TITLE
"window is not defined" error (SvelteKit)

### ISSUE THREAD

**Issue by tomgenoni** [thecodejack/svelte-file-dropzone#81]
Using the default example in the docs I'm getting:

```
500
window is not defined
ReferenceError: window is not defined
    at Dropzone.svelte:293:10
    at run (/Users/tom/Sites/slack-demo/node_modules/svelte/internal/index.js:22:12)
    at Array.forEach (<anonymous>)
    at run_all (/Users/tom/Sites/slack-demo/node_modules/svelte/internal/index.js:28:9)
    at Object.render (/Users/tom/Sites/slack-demo/node_modules/svelte/internal/index.js:1693:13)
    at render_response (file:///Users/tom/Sites/slack-demo/node_modules/@sveltejs/kit/dist/ssr.js:561:28)
    at async respond$1 (file:///Users/tom/Sites/slack-demo/node_modules/@sveltejs/kit/dist/ssr.js:1403:4)
    at async render_page (file:///Users/tom/Sites/slack-demo/node_modules/@sveltejs/kit/dist/ssr.js:1480:19)
    at async resolve (file:///Users/tom/Sites/slack-demo/node_modules/@sveltejs/kit/dist/ssr.js:1743:10)
    at async respond (file:///Users/tom/Sites/slack-demo/node_modules/@sveltejs/kit/dist/ssr.js:1722:10)
```

Here's my `package.json`

```
{
  "name": "demo",
  "version": "0.0.1",
  "scripts": {
    "dev": "svelte-kit dev",
    "build": "svelte-kit build",
    "preview": "svelte-kit preview",
    "lint": "prettier --ignore-path .gitignore --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
    "format": "prettier --ignore-path .gitignore --write --plugin-search-dir=. ."
  },
  "devDependencies": {
    "@sveltejs/kit": "next",
    "eslint": "^7.32.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-svelte3": "^3.2.1",
    "prettier": "^2.4.1",
    "prettier-plugin-svelte": "^2.4.0",
    "svelte": "^3.42.6"
  },
  "type": "module",
  "dependencies": {
    "svelte-file-dropzone": "^0.0.15"
  }
}
```

Likely an issue with SvelteKit.

**Comment by thecodejack** [comment]
probably something changed with `sveltekit`. Any pointers to debug. Have very minimal time to fix issue for me.

**Comment by thecodejack** [comment]
Tried with sveltekit....didn't see any error. Added sveltekit example as well
https://github.com/thecodejack/svelte-file-dropzone/tree/master/examples/sveltekit-js

**Comment by thecodejack** [comment]
By any chance is your app using TS?

**Comment by skanel** [comment]
it is the same for my case.
I am using sveltkit too, but the problem raise only when we try to refresh the page.

**Comment by AbramovYuri** [comment]
```
window is not defined

ReferenceError: window is not defined
    at Dropzone.svelte:293:10
    at run (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:22:12)
    at Array.forEach (<anonymous>)
    at run_all (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:28:9)
    at Object.render (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:1711:13)
    at render_response (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:561:28)
    at async respond$1 (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1377:4)
    at async render_page (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1454:19)
    at async resolve (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1717:10)
    at async respond (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1696:10)
```
I also have such an error, adapter static is connected in svelte-kit and ssr is enabled

**Comment by thecodejack** [comment]
ok let me add a condition to avoid same.

**Comment by thecodejack** [comment]
please check with `svelte-file-dropzone@0.0.16-dev.0` and let me know if that fixes the issue

**Comment by AbramovYuri** [comment]
```
window is not defined

ReferenceError: window is not defined
    at Dropzone.svelte:293:10
    at run (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:22:12)
    at Array.forEach (<anonymous>)
    at run_all (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:28:9)
    at Object.render (D:\Work\OpenServer\domains\proks-form-frontend\node_modules\svelte\internal\index.js:1711:13)
    at render_response (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:561:28)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async respond$1 (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1377:4)
    at async render_page (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1454:19)
    at async resolve (file:///D:/Work/OpenServer/domains/proks-form-frontend/node_modules/@sveltejs/kit/dist/ssr.js:1717:10)
```
    `"svelte-file-dropzone": "0.0.16-dev.0",`
It didn't help, the error persisted(

**Comment by thecodejack** [comment]
is it possible to provide a reproducible link? like a repo ?

**Comment by AbramovYuri** [comment]
https://github.com/AbramovYuri/dropzone-example.git

Delete this code to reproduce the error
![изображение](https://user-images.githubusercontent.com/43909379/148122739-f216de20-f016-42cc-823d-3d333c957e09.png)

**Comment by skanel** [comment]
i am using sveltekit , this solution works for me, thanks

**Comment by thecodejack** [comment]
Thanks for reproducer @AbramovYuri

**Comment by thecodejack** [comment]
ok...looks like this can be sveltekit issue also. I am guessing code inside `onDestroy` expected to run in browser just like `onMount`. I remember moving global `window` code to life events just to make it SSR compatible via `sapper`. Should have worked same with `sveltekit`.

Anyways I will also debug via reproducer code shared by @AbramovYuri  to check I can fix it in anyway that this is SSR compatible.

**Comment by thecodejack** [comment]
`svelte-file-dropzone@1.0.0-dev.1` should fix the issue. But the component is using sveltekit internals(https://kit.svelte.dev/docs#modules-$app-env). This is breaking storybook. Working on fixing same. 

I am at present fine to use `sveltekit` internal API to add SSR support, as long as it doesn't break storybook docs & svelte REPL. 

Let know if you feel different.

**Comment by thecodejack** [comment]
Raised issue with REPL to add support https://github.com/sveltejs/sites/issues/262

**Comment by AbramovYuri** [comment]
thank you, it helped me, now I don't need to disable ssr anymore, the error is gone)

**Comment by PlkMarudny** [comment]
This line:

`import { browser } from "$app/env"`

is incompatible with a regular Svelte application (when no SvelteKit is used), it throws the error during compilation.

**Comment by thecodejack** [comment]
yes..I am not going to publish a standard release this mostly until there is a package exposed.

**Comment by thecodejack** [comment]
It might disappoint for few people here. At present I have no option but release upgraded version which unfortunately won't have SSR support on Sveltekit. Please keep using `svelte-file-dropzone@1.0.0-dev.1` until we have proper solution. I am trying to explore final possibility of breaking this into different packages so that sveltekit will have SSR enabled support.


What is the issue?
At present, onDestroy is not part of Sveltekit SSR lifecycle. And it's not intended to be. And with that we have to add hack. https://github.com/sveltejs/kit/issues/945 

But that means we are importing dynamic module $app which is not available in non-sveltekit apps. I tried asking for shims but no luck so far.
https://github.com/sveltejs/sites/issues/262
https://github.com/sveltejs/kit/issues/1485

**Comment by thecodejack** [comment]
This should have been fixed with v1 release.

**Comment by Tal500** [comment]
fixed in #105
